RFID do picosdk_rewrite

Fellapp opened the pull request.

Fellapp committed:
RFID

Fellapp committed:
getUid

igoro00 committed:
zmiana pinów na prawdziwe

igoro00 committed:
usunięcie castowania na void*

Fellapp committed:
CS na 0

Fellapp committed:
rfid

Fellapp committed:
Działa pobieranie UUID

Fellapp committed:
piny z pindefs

Fellapp committed:
Dodanie attachInterrupt i detachInterrupt

Fellapp committed:
IRQ

Fellapp committed:
Merge branch 'picosdk_rewrite' of https://github.com/igoro00/wbudy into jakub

Fellapp committed:
dodanie tasków, dodanie konstruktora bezparametrowego, dodanie freertos do CMake

Fellapp committed:
Tworzenie tasków w main, init RFID w main, rfid w Context

igoro00 commented:
@Fellapp 

- Tworzenie tasków powinno być pod koniec inita (patrz: `WbudyButton::init`)
- Nie zawsze potrzebujemy callbacka (wsm przez większość czasu go nie potrzebujemy) więc nie dodawaj go do `setupPins()`. Ja go później dodam w `sLobby.cpp` kiedy będe go potrzebował
- Stwórz nowy semafor binarny który będzie blokował `tPing` jeśli nie ma callbacka podpiętego i bierz/oddawaj ten semafor w `attachInterrupt()` i `detachInterrupt()`. Pamiętaj o oddawaniu semafora po zapingowaniu (żeby `tPing` mógł cały czas działać w pętli).
- Zmień te nazwy `attachInterrupt()` i `detachInterrupt` bo to nie jest interrupt. W `WbudyButton` jest `setOnPressed` wieć u ciebie może być np. `setOnScanned`. Może to być jedna funkcja która usunie callback jak przekażesz `NULL` jako argument albo doda jak `!= NULL`
- Na koniec napraw znowu merge conflicty. (nie dodawaj starych, niepotrzebnych headerów do include'a)

Fellapp committed:
tworzenie tasków pod koniec inita,
usunięcię callbacka (attachInterrupt()) z setupPins() z main(), dodanie bibkliotek:

Fellapp committed:
dodanie mutexa do blokowania tPing

Fellapp committed:
zmiana attachInterrupt() i detachInterrupt() na setOnScanned()

Fellapp committed:
Merge branch 'picosdk_rewrite' of https://github.com/igoro00/wbudy into jakub, usunięcię starego headera

igoro00 commented:
Co jeśli w czasie gdy właśnie przetwarzasz RFID (smphr wzięty) będziesz chciał odłączyć CB?

igoro00 commented:
Sprawdzałeś czy to się kompiluje?

Usunąłeś ostatnim(chyba) commitem RFID z Context a w main.cpp używasz `ctx.rfid`.

No i chyba ogólnie nadpisałeś sobię tego maina bo masz tam attachInterrupt

Fellapp commented:
> Co jeśli w czasie gdy właśnie przetwarzasz RFID (smphr wzięty) będziesz chciał odłączyć CB?

Trzeba sprawdzić czy w tReadCard callback nadal istnieje

igoro00 commented on platformio.ini:
To chyba do usunięcia?

igoro00 commented on src/main.cpp:
```diff
@@ -46,6 +46,26 @@ void setupPins(){
+		RFID_MOSI
+	);
+
+	ctx.rfid.attachInterrupt([](uint32_t uid) {
+		printf("[RFID] Card detected with UID: %08X\n", uid);
+		ctx.cardUID = uid;
+		if (ctx.gameState == GameState::LOBBY || ctx.gameState == GameState::GAME) {
+			if (ctx.game) {
+				ctx.game->setPlayer(true, uid);
+			}
+		}
+	});
```
Chyba sobie to nadpisałeś

Fellapp replied:
Tak

igoro00 commented on test/README:
Huh?

igoro00 commented:
> > Co jeśli w czasie gdy właśnie przetwarzasz RFID (smphr wzięty) będziesz chciał odłączyć CB?
> 
> Trzeba sprawdzić czy w tReadCard callback nadal istnieje

No ok, i sprawdzasz to (dobrze) ale dalej będzie pingowal na darmo

Fellapp committed:
usunięcie z maina attachInterrupt

Fellapp committed:
sprawdzenie czy callback nie został odłączony w trakcie działania

Fellapp commented:
> > > Co jeśli w czasie gdy właśnie przetwarzasz RFID (smphr wzięty) będziesz chciał odłączyć CB?
> > 
> > 
> > Trzeba sprawdzić czy w tReadCard callback nadal istnieje
> 
> No ok, i sprawdzasz to (dobrze) ale dalej będzie pingowal na darmo

> > > Co jeśli w czasie gdy właśnie przetwarzasz RFID (smphr wzięty) będziesz chciał odłączyć CB?
> > 
> > 
> > Trzeba sprawdzić czy w tReadCard callback nadal istnieje
> 
> No ok, i sprawdzasz to (dobrze) ale dalej będzie pingowal na darmo

To dodałem w pętli tPing:
if (self->_callback == nullptr) {
              continue;
          }

igoro00 reviewed (approved):
LGTM
